crates/interpreter/src/valid.rs

```diff
@@ -826,3 +898,67 @@ impl<'a, 'm> Expr<'a, 'm> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use alloc::format;
+
+    use proptest::prelude::*;
+
+    use super::bit_field::*;
+    use crate::Error;
+
+    proptest! {
+        #[test]
+        fn signed_field_round_trip(value in -8i32..8) {
+            let mask = 0b1111000;
+            let field = into_signed_field(mask, value).unwrap();
+            let recovered = from_signed_field(mask, field);
+            prop_assert_eq!(value, recovered);
+        }
+
+        #[test]
+        fn unsigned_field_round_trip(value in 0u32..8) {
+            let mask = 0b111000;
+            let field = into_field(mask, value).unwrap();
+            let recovered = from_field(mask, field);
+            prop_assert_eq!(value, recovered);
+        }
+    }
+
+    #[test]
+    #[cfg(debug_assertions)]
+    fn into_field_error_debug() {
+        // This mask has 3 bits set, so it can only represent values 0-7.
+        let mask = 0b111000;
+        let value = 8;
+
+        let result = into_field(mask, value);
+        assert!(result.is_err());
+        assert!(matches!(result.unwrap_err(), Error::Unsupported(_)));
+    }
```

Comment:
can be simplified to
```suggestion
    fn into_field_error() {
        // The value 8 cannot be represented on 3 bits.
        assert!(matches!(into_field(0b1110000, 8), Err(Error::Unsupported(_))));
    }
```